zipline/data/us_equity_pricing.py

```diff
@@ -760,15 +759,33 @@ class PanelDailyBarReader(DailyBarReader):
         The first trading day in the dataset.
     """
     @preprocess(panel=call(verify_indices_all_unique))
-    def __init__(self, calendar, panel):
+    def __init__(self, trading_calendar, panel, data_frequency='daily'):
 
         panel = panel.copy()
         if 'volume' not in panel.minor_axis:
             # Fake volume if it does not exist.
             panel.loc[:, :, 'volume'] = int(1e9)
 
-        self.first_trading_day = panel.major_axis[0]
-        self._calendar = calendar
+        self._trading_calendar = trading_calendar
+        self.first_trading_day = trading_calendar.minute_to_session_label(
+            panel.major_axis[0]
+        )
+        last_trading_day = trading_calendar.minute_to_session_label(
+            panel.major_axis[-1]
+        )
+
+        self._sessions = trading_calendar.sessions_in_range(
+            self.first_trading_day,
+            last_trading_day
+        )
+
+        if data_frequency == 'daily':
+            self._calendar = self._sessions
+        elif data_frequency == 'minute':
```

Comment:
Can we add an `ensure_element(data_frequency={'daily', 'minute'})` so that users cannot pass invalid frequencies?